server/manager/src/main/java/org/apache/accumulo/manager/upgrade/UpgradeCoordinator.java

```diff
@@ -179,15 +187,21 @@ public synchronized Future<Void> upgradeMetadata(ServerContext context,
           .submit(() -> {
             try {
               for (int v = currentVersion; v < AccumuloDataVersion.get(); v++) {
-                log.info("Upgrading Root from data version {}", v);
-                upgraders.get(v).upgradeRoot(context);
+                log.info("Upgrading Root from data version {} target version {}", v,
+                    AccumuloDataVersion.get());
+                if (upgraders.get(v) != null) {
+                  upgraders.get(v).upgradeRoot(context);
+                }
               }
 
               setStatus(UpgradeStatus.UPGRADED_ROOT, eventCoordinator);
 
               for (int v = currentVersion; v < AccumuloDataVersion.get(); v++) {
-                log.info("Upgrading Metadata from data version {}", v);
-                upgraders.get(v).upgradeMetadata(context);
+                log.info("Upgrading Metadata from data version {} target version {}", v,
+                    AccumuloDataVersion.get());
+                if (upgraders.get(v) != null) {
+                  upgraders.get(v).upgradeMetadata(context);
+                }
```

Comment:
Why the null checks here? Under what circumstances would we find an upgrader to be null? If that happens, we don't want to completely ignore it, but we should fail hard. That's not expected.

Reply:
Addressed with b3ca1d9b65

Reply:
I think you should replace lines 202 - 204 with `upgraders.get(v).upgradeMetadata(context);`. You added the `requireNonNull` check in the commit you mentioned, so it shouldn't be null. If it null here, then this code will hide it and move on.